torch/_dynamo/guards.py

```diff
@@ -1300,20 +1317,34 @@ def get_and_maybe_log_recompilation_reason(
 
     code = frame.f_code
 
+    # at least one of "recompiles" or "recompiles_verbose" is enabled
     do_recompiles_log = (
-        is_guard_failure_reporting_enabled()
-        and recompiles_log.isEnabledFor(logging.DEBUG)
+        is_recompiles_enabled() and recompiles_log.isEnabledFor(logging.DEBUG)
+    ) or (
+        is_recompiles_verbose_enabled()
+        and recompiles_verbose_log.isEnabledFor(logging.DEBUG)
     )
 
     if do_recompiles_log or config.error_on_recompile:
-        failures = "\n".join(reasons)
-        guard_failure_details = f"triggered by the following guard failure(s):\n{textwrap.indent(failures, '- ')}"
+        if is_recompiles_verbose_enabled():
+            failures = "\n\n".join(
+                f"guard {i} failures:\n" + textwrap.indent(reason, "- ")
+                for i, reason in enumerate(reasons)
+            )
+        else:
+            failures = textwrap.indent("\n".join(reasons), "- ")
+        guard_failure_details = (
+            f"triggered by the following guard failure(s):\n{failures}"
+        )
         message = (
             f"Recompiling function {code.co_name} in {code.co_filename}:{code.co_firstlineno}\n"
             f"{textwrap.indent(guard_failure_details, '    ')}"
         )
         if do_recompiles_log:
-            recompiles_log.debug(message, stack_info=True)
+            if recompiles_log.isEnabledFor(logging.DEBUG):
```

Comment:
This is kind of weird, why are you using an if statement on isEnabledFor on recompiles_log here?

Reply:
I think it should all go to verbose if verbose is enabled, and go to recompile if it is enabled. I.e. verbose will override the recompiles.

Reply:
So verbose should be checked first, no?

Reply:
I think it would be more complicated to log partially to regular `recompiles` and partially to `recompiles_verbose` - so I'm just picking one of them based on what's enabled.

Reply:
yes